### DISCUSSION TITLE
How to build Superset as Python package with frontend from source?

### DISCUSSION THREAD

**Discussion by thinkh** [apache/superset#23183] [Q&A / Help]
Hi,
I'd like to create a Python package which includes the frontend from the source code. I saw that I can build the assets following the [frontend section in the contributing guide](https://github.com/apache/superset/blob/master/CONTRIBUTING.md#frontend).

``` 
# From the root of the repository
cd superset-frontend

# Install dependencies from `package-lock.json`
npm ci

# Build the production assets, CSS/JSS minified and optimized
npm run build
```

I know that I need to move/copy the assets to `/superset/static/assets`. Next I would need to build the Python package as a next step. Are there any instructions how to build the Python package? What would be the steps?

**Comment by sebastianliebscher** [comment]


**Reply by seven07ve** [reply]
# Build javascript assets
cd superset-frontend
Where ??? is asking

**Reply by linggangcong** [reply]
source code of superset

**Comment by ghost** [comment]
I think the confusion is that the Installing From Scratch documentation has you install via pip: https://superset.apache.org/docs/installation/installing-superset-from-scratch

But when installing with pip you don't find the superset-frontend folder and can't successfully follow the later part of the doc.

**Reply by robertobasile84** [reply]
I'm having the same issue. So, how do we install it from scratch? I am at the point where I cannot move on since I don't have that folder

**Reply by rteja0209** [reply]
Even I am stuck at this point
